pkg/compose/up.go

```diff
@@ -88,6 +96,27 @@ func (s *composeService) Up(ctx context.Context, project *types.Project, options
 		}
 		for {
 			select {
+			case event := <-kEvents:
+				switch key := event.Key; key {
+				case keyboard.KeyCtrlC:
+					keyboard.Close()
+					formatter.KeyboardInfo.ClearInfo()
+					gracefulTeardown()
```

Comment:
I wonder about the impact doing so as context is not cancelled.

Reply:
I'm not sure. We rely on `<-ctx.Done()` in a few place to stop goroutines after Ctrl+C, typically to stop collecting logs, watch container events, etc. I assume after the application stopped those will be released anyway, but this might bring some new race conditions.

Reply:
I think it should be okay here (although it makes me think the graceful termination code/`doneCh`/`ctx.Done()` handling could use a refactor).

I wonder if cancelling the context here would be better.

Also I wonder, since we're already capturing `SIGINT` above (when we do `signal.Notify(signalChan, syscall.SIGINT, syscall.SIGTERM)`, if the new `keyboard` stuff is going to interfere with that.

Reply:
Changed the way we handle the tear down. I am sending a SIGTERM signal to signalChan